.github/workflows/build-release-pip-package.yml

```diff
@@ -1,12 +1,8 @@
 name: Build and Release pip package
 
 on:
-  workflow_dispatch:
-    inputs:
-      branch:
-        description: 'Branch to checkout'
-        required: true
-        default: 'main'
+  release:
+    types: [published, prereleased]
```

Comment:
AFAIK, this will trigger twice for pre-releases if we first created a draft pre-release and then publish it. I think "prereleased" event is not necessary?

Maybe the best hook would be that we trigger when a draft release is created so that when we publish it, it already has the asset. This would mean to trigger on "created".

Reply:
As per the documentation, the prereleased type will not trigger for pre-releases published from draft releases, but the published type will trigger. It will only trigger once.
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release

Yes, that's the best way. When we create the draft release, the workflow should attach the Python wheel to it so we can test it from the draft release itself. Unfortunately, workflows can't be triggered for the created, edited, or deleted activity types for draft releases.

One option is to run a manual workflow once the draft release is created.?